plugins/volumeFilter.go

```diff
@@ -175,74 +175,95 @@ func (f *volumeFilter) Apply(ops []txnbuild.Operation, sellingOffers []hProtocol
-					newAmountString = ", newAmountString = " + opToReturn.Amount
-				}
-			}
-			log.Printf("volumeFilter:  selling (base units), price=%.8f amount=%.8f, keep = (projectedSoldInBaseUnits) %.7f <= %.7f (config.BaseAssetCapInBaseUnits): keepSellingBase = %v%s", sellPrice, amountValueUnitsBeingSold, projectedSoldInBaseUnits, *lp.baseAssetCapInBaseUnits, keepSellingBase, newAmountString)
-		} else {
-			keepSellingBase = true
-		}
+	// extracts from base or quote side, depending on filter
+	otb, tbb, cap, e := extractFilterInputs(dailyOTB, dailyTBBAccumulator, lp)
+	if e != nil {
+		return nil, fmt.Errorf("could not extract filter inputs from filter: %s", e)
+	}
 
-		if lp.baseAssetCapInQuoteUnits != nil {
-			projectedSoldInQuoteUnits := *dailyOTB.BaseAssetCapInQuoteUnits + *dailyTBBAccumulator.BaseAssetCapInQuoteUnits + (newAmountBeingSold * sellPrice)
-			keepSellingQuote = projectedSoldInQuoteUnits <= *lp.baseAssetCapInQuoteUnits
-			newAmountString := ""
-			if lp.mode == volumeFilterModeExact && !keepSellingQuote {
-				newAmount := (*lp.baseAssetCapInQuoteUnits - *dailyOTB.BaseAssetCapInQuoteUnits - *dailyTBBAccumulator.BaseAssetCapInQuoteUnits) / sellPrice
-				if newAmount > 0 {
-					newAmountBeingSold = newAmount
-					opToReturn.Amount = fmt.Sprintf("%.7f", newAmountBeingSold)
-					keepSellingQuote = true
-					newAmountString = ", newAmountString = " + opToReturn.Amount
-				}
-			}
-			log.Printf("volumeFilter: selling (quote units), price=%.8f amount=%.8f, keep = (projectedSoldInQuoteUnits) %.7f <= %.7f (config.BaseAssetCapInQuoteUnits): keepSellingQuote = %v%s", sellPrice, amountValueUnitsBeingSold, projectedSoldInQuoteUnits, *lp.baseAssetCapInQuoteUnits, keepSellingQuote, newAmountString)
-		} else {
-			keepSellingQuote = true
-		}
+	// if projected is under the cap, update the tbb and return the original op
+	projected := otb + tbb + offerAmount*capPrice
+	if projected <= cap {
+		*dailyTBBAccumulator.BaseAssetCapInBaseUnits += offerAmount
+		*dailyTBBAccumulator.BaseAssetCapInQuoteUnits += offerAmount * offerPrice
+		return op, nil
+	}
 
-		if keepSellingBase && keepSellingQuote {
-			// update the dailyTBB to include the additional amounts so they can be used in the calculation of the next operation
-			*dailyTBBAccumulator.BaseAssetCapInBaseUnits += newAmountBeingSold
-			*dailyTBBAccumulator.BaseAssetCapInQuoteUnits += (newAmountBeingSold * sellPrice)
-			return opToReturn, nil
-		}
-	} else {
-		// TODO buying side - we need to implement this to support buy side filters; extract common logic from the above sell side case
+	if lp.mode != volumeFilterModeExact {
+		return nil, nil
 	}
 
-	// we don't want to keep it so return the dropped command
-	return nil, nil
+	// if exact mode and with remaining capacity, update the op amount and return the op
+	// else, return nil
+	remainingCapacity := (cap - otb - tbb) / capPrice
+	if remainingCapacity <= 0 {
+		return nil, nil
+	}
+
+	offerAmount = remainingCapacity
```

Comment:
is it correct to call this `offerBaseAmount` because we divide by `capPrice`? if so then let's use that name to be more explicit since we have the context on it now but not when we will revisit the code next in the future

Reply:
PS: we should try to avoid reusing variables where possible. Using new variable names as a general practice makes the code more readable since developers don't need to remember what this variable was assigned to previously and if we ever reuse it again.

A good name for this to keep it separate from `offerAmount` would be either `newOfferAmount` or `newOfferBaseAmount`.

Reply:
PPS: that being said, we probably don't need a separate variable for `offerAmount` or `offerAmountBase` since it's the same thing as `remainingCapacity` which we can probably rename to `remainingCapacityBase`. alternatively we can rename `remainingCapacity` to `offerAmount` or `offerAmountBase` accordingly.
i.e. we don't do anything to `offerAmount` here so this is effectively just a rename which may not be required

Reply:
Yeah, this all makes sense. I think you're right that we don't need to assign this to `offerAmount`. I'm just going to keep it as `remainingCapacity`.